src/audio/copier.c

```diff
@@ -745,27 +745,25 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 	 * for a short time when the second pipeline already started
 	 * and the first one is not ready yet along with sink buffers params
 	 */
-	list_for_item(src_list, &dev->bsource_list) {
-		int  i, j;
-
-		source = container_of(src_list, struct comp_buffer, sink_list);
-		if (source->hw_params_configured == false) {
-			j = IPC4_SRC_QUEUE_ID(source->id);
-			source->stream.channels = cd->out_fmt[j].channels_count;
-			source->stream.rate = cd->out_fmt[j].sampling_frequency;
-			audio_stream_fmt_conversion(cd->out_fmt[j].depth,
-						    cd->out_fmt[j].valid_bit_depth,
-						    &source->stream.frame_fmt,
-						    &source->stream.valid_sample_fmt,
-						    cd->out_fmt[j].s_type);
-
-			source->buffer_fmt = cd->out_fmt[j].interleaving_style;
-
-			for (i = 0; i < SOF_IPC_MAX_CHANNELS; i++)
-				source->chmap[i] = (cd->out_fmt[j].ch_map >> i * 4) & 0xf;
-
-			source->hw_params_configured = true;
-		}
+	source = list_first_item(&dev->bsource_list, struct comp_buffer, sink_list);
```

Comment:
can you update the comment above to change "update each source" to "update the source"?

Reply:
sure